Plugins/DotNET/DotNETExports.cpp

```diff
@@ -428,61 +461,93 @@ static int32_t ClosureActionDoCommand(uint32_t oid, uint64_t eventId)
     return 0;
 }
 
-static void nwnxSetFunction(const char *plugin, const char *function)
+static void NWNXSetFunction(const char* plugin, const char* function)
 {
     s_nwnxActivePlugin = plugin;
     s_nwnxActiveFunction = function;
 }
-static void nwnxPushInt(int32_t n)
+
+static void NWNXPushInt(int32_t n)
 {
     Events::Push(n);
 }
-static void nwnxPushFloat(float f)
+
+static void NWNXPushFloat(float f)
 {
     Events::Push(f);
 }
-static void nwnxPushObject(uint32_t o)
+
+static void NWNXPushObject(uint32_t o)
 {
     Events::Push((ObjectID)o);
 }
-static void nwnxPushString(const char *s)
+
+static void NWNXPushString(const char* s)
 {
     Events::Push(String::FromUTF8(s));
 }
-static void nwnxPushEffect(CGameEffect *e)
+
+static void NWNXPushRawString(const char* s)
+{
+    Events::Push<std::string>(s);
+}
+
+static void NWNXPushEffect(CGameEffect* e)
 {
     Events::Push(e);
 }
-static void nwnxPushItemProperty(CGameEffect *ip)
+
+static void NWNXPushItemProperty(CGameEffect* ip)
 {
     Events::Push(ip);
 }
-static int32_t nwnxPopInt()
+
+static int32_t NWNXPopInt()
 {
     return Events::Pop<int32_t>().value_or(0);
 }
-static float nwnxPopFloat()
+
+static float NWNXPopFloat()
 {
     return Events::Pop<float>().value_or(0.0f);
 }
-static uint32_t nwnxPopObject()
+
+static uint32_t NWNXPopObject()
 {
     return Events::Pop<ObjectID>().value_or(Constants::OBJECT_INVALID);
 }
-static const char* nwnxPopString()
+
+static const char* NWNXPopString()
 {
     auto str = Events::Pop<std::string>().value_or(std::string{""});
     return strdup(String::ToUTF8(str).c_str());
 }
-static CGameEffect* nwnxPopEffect()
+
+static const char* NWNXPopRawString()
+{
+    auto value = Events::Pop<std::string>();
+    if (value.has_value())
+    {
+      static auto retVal = value.value();
+      return retVal.c_str();
+    }
+    else
+    {
+      return nullptr;
```

Comment:
As above, `""` instead of `nullptr`. But this branch goes away if you use `value_or()`

Reply:
As above, the intention is that null pointers are handled by the managed code.